packages/js/src/ui/components/Renderer.tsx

```diff
@@ -75,13 +79,14 @@ export const Renderer = (props: RendererProps) => {
           <FocusManagerProvider>
             <InboxProvider tabs={props.tabs}>
               <CountProvider>
-                <For each={[...props.nodes]}>
-                  {([node, component]) => {
+                <For each={nodes()}>
+                  {(node) => {
+                    const novuComponent = createMemo(() => props.nodes.get(node)!);
```

Comment:
However, the Solid component props might change, which is why we should memoize it and update the signal when it has been changed.